### PR TITLE
Add a warning for large image sizes

### LINKED ISSUE #395: Add a warning for large image sizes
Display a warning if there are images that are larger than 500 KB, to help remind authors that large images can slow the website down.

### PR DESCRIPTION
**Pull request by carolinan**
Fixes #395
Displays a warning if there are images that are larger than 500 KB.

### Screenshot:
image

### How to test:
1. Run Theme Check on any theme with an image file larger than 500 KB.
2. See the warning

### CODE REVIEW

**Comment by kafleg** [comment]
I got this error while testing. 

Fatal error: Uncaught Error: Call to undefined function exif_imagetype() in E:\laragon\www\reviews\wp-content\plugins\theme-check\checks\class-image-size-check.php:32 Stack trace: #0 E:\laragon\www\reviews\wp-content\plugins\theme-check\checkbase.php(114): Image_Size_Check->check(Array, Array, Array) #1 E:\laragon\www\reviews\wp-content\plugins\theme-check\checkbase.php(83): run_themechecks(Array, Array, Array, Array) #2 E:\laragon\www\reviews\wp-content\plugins\theme-check\main.php(27): run_themechecks_against_theme(Object(WP_Theme), 'creativ-academy') #3 E:\laragon\www\reviews\wp-content\plugins\theme-check\theme-check.php(66): check_main('creativ-academy') #4 E:\laragon\www\reviews\wp-includes\class-wp-hook.php(303): ThemeCheckMain->themecheck_do_page('') #5 E:\laragon\www\reviews\wp-includes\class-wp-hook.php(327): WP_Hook->apply_filters('', Array) #6 E:\laragon\www\reviews\wp-includes\plugin.php(470): WP_Hook->do_action(Array) #7 E:\laragon\www\reviews\wp-admin\admin.php(259): do_action('appearance_page...') #8 E:\la in E:\laragon\www\reviews\wp-content\plugins\theme-check\checks\class-image-size-check.php on line 32

**Comment by carolinan** [comment]
It seems it needs a PHP extension to be installed then.

I found this WP function, which I did not know already existed: https://developer.wordpress.org/reference/functions/wp_get_image_mime/
I will try with that, it already checks for exif_imagetype and getimagesize.

File: checks/class-image-size-check.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	public function check( $php_files, $css_files, $other_files ) {
+
+		checkcount();
+
+		foreach ( $other_files as $other_key => $otherfile ) {
+			/*
+			* Check if the file is an image.
+			* Silence read error if the file is too small. @see https://www.php.net/manual/en/function.exif-imagetype.php#79283.
+			*/
+			if ( @wp_get_image_mime( $other_key ) !== false ) {
```
Comment (carolinan) [inline checks/class-image-size-check.php:35]: Can we live with the silence here or do I need to rewrite it to do the file size check _before_ checking if it is actually an image?

**Comment by kafleg** [comment]
Worked perfectly now!

This is the warning message displayed. 

`WARNING: IMG_8091.jpg is 2469 KB large. Large file sizes have a negative impact on website performance and loading time. Compress images before using them.
`

**Pull request merged**